### ISSUE TITLE
Testing the Core

### ISSUE THREAD

**Issue by jonaseberledmind** [TYPO3-Documentation/TYPO3CMS-Guide-ContributionWorkflow#93]
Maybe I am missing something important, but I have issues following the guide https://github.com/TYPO3-Documentation/TYPO3CMS-Guide-ContributionWorkflow/blob/master/Documentation/Testing/Index.rst

It should be noted IMHO that the steps from https://docs.typo3.org/typo3cms/ContributionWorkflowGuide/Setup/SetupTypo3.html?highlight=yarn are necessary first.

The docker image ‹typo3gmbh/php72:latest› has no file /sbin/my_init - so the commands under "docker run" give an error about it.

There is also no ‹bin/chromedriver› in the docker image. 

- I am not sure if this is expected (then it should be noted) or if it is an oversight in the documentation or a bug in the underlying Dockerfiles

Probably I missed something. Maybe I am using it wrong. I wanted to run the core tests locally.

**Comment by liayn** [comment]
@lolli42 can maybe tell more about the docker image

**Comment by sypets** [comment]
> It should be noted IMHO that the steps from 
> https://docs.typo3.org/typo3cms/ContributionWorkflowGuide/Setup/SetupTypo3.html?highlight=yarn are > necessary first.

The contribution guide in general assumes, that the previously described steps have been followed, but it is better to be clear: I will add a section "prerequisites".

> The docker image ‹typo3gmbh/php72:latest› has no file /sbin/my_init - so the commands under 
> "docker run" give an error about it.

Yes, I get the same error. It appears there is currently some work in progess with the docker images and bamboo. I have asked in the #bamboo channel on https://typo3.slack.org and hopefully someone will get back to you.

As for the chromedriver, this will probably be addressed as well. 

Please be patient. I am sure this will be resolved soon. Your intentions to run the tests is admirable and thanks for taking the time to raise the issue.

**Comment by jonaseberledmind** [comment]
Thank you. 
I helped myself and put together a docker-compose.yml, a script to build Typo3 and 2 scripts to run unit- and functional tests. 
I could even get rid of the extra scripts and include the entry points in the docker-compose.yml and get rid of the local dependency on yarn and cmdtest to run the tests  - what do you think, would that help? 

I think docker-compose is such a beautiful tool to set up an easily reproducible environment. 

https://gist.github.com/jonaseberledmind/d9919794543f4430509f1a3a310475b6

**Comment by sypets** [comment]
The testing workflow is being refactored (see https://review.typo3.org/c/57789/).

We should update the testing page to reflect this and add some additional information:

* [ ] `typo3temp/var/tests` (or probably entire `typo3temp` dir) should be excluded in IDE (e.g. PhpStorm: Settings:Directories:Excluded)
* [ ] ...

**Comment by jonaseberledmind** [comment]
When yarn gets packaged inside the extended runTest.sh-command (https://review.typo3.org/c/57789/),
the usefulness could be multiplied by mentioning the docker-compose based solution here https://docs.typo3.org/typo3cms/ContributionWorkflowGuide/Setup/SetupTypo3.html - no need to locally install yarn any more for normal patch testing.

**Comment by sypets** [comment]
@jonaseberledmind Good idea. Or document that on https://docs.typo3.org/typo3cms/ContributionWorkflowGuide/Setup/Prerequisites.html. Feel free to add another issue here for that. 

About yarn: See also #55
